### PR TITLE
Add IP white-listing functionallity

### PR DESCRIPTION
**Pull request by RONTheCookie**
This is my first time contributing to a FOSS project, so please tell me if I'm doing anything wrong (:

### CODE REVIEW

**Comment by manast** [comment]
Thanks for the contribution. I have been on vacation and did not have time to make a review.

File: lib/proxy.js [hunk 1]
```diff
@@ -272,6 +282,16 @@ ReverseProxy.prototype.setupHttpsProxy = function (proxy, websocketsUpgrade, log
     var src = _this._getSource(req);
     var httpProxyOpts = Object.assign({}, _this.opts.httpProxy);
 
+    if (opts.whitelistedIps) {
```
Comment (manast) [inline lib/proxy.js]: You should refactor this code sin it is the same as the one starting in line 182.

**Comment by RONTheCookie** [comment]
Busy right now but I'll get around to doing it

**Comment by RONTheCookie** [comment]
Closing for now, can reopen if anyone wants to take over

**Pull request closed without merging**